CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+# Contributing
+We hope to steer development of Connaisseur from demand of the community and are excited about direct contributions to improve the tool!
+
+The following guide is to meant to help you get started with contributing to Connaisseur.
+
+We are committed to positive interactions between all contributors of the project. To ensure this, please follow the [Berlin Code of Conduct](https://berlincodeofconduct.org/) in all communications.
+
+## Contents
+- [Discuss Problems, Raise Bugs and Propose Feature Ideas](#discuss-problems-raise-bugs-and-propose-feature-ideas)
+- [Contribute to Source Code](#contribute-to-source-code)
+  * [Setup the Environment](#setup-the-environment)
+  * [Test Changes](#test-changes)
+  * [Signed Commits and Pull Requests](#signed-commits-and-pull-requests)
+  * [Semantic and Conventional Commits](#semantic-and-conventional-commits)
+- [Enjoy!](#enjoy)
+
+## Discuss Problems, Raise Bugs and Propose Feature Ideas
+We are happy you made it here! If you want to raise any bugs you found, discuss issues from using Connaisseur in your own projects or have ideas for new features to be implemented, please [create an issue](https://github.com/sse-secure-systems/connaisseur/issues/new) with an informative title and description.
+
+While issues are a great way to discuss problems, bugs and new features, a direct proposal via a pull request can sometimes say more than a thousand words. So be bold and contribute to the code as described in the [next section](#contribute-to-source-code)!
+
+In case you require a more private communication, you can reach us via [connaisseur@securesystems.de](mailto:connaisseur@securesystems.de).
+
+## Contribute to Source Code
+The following steps will help you make code contributions to Connaisseur and ensure good code quality and workflow. This includes the following steps:
+1. **Setup your environment**: Setup up your local environment to best interact with the code. Further information is given in the subsection below.
+2. **Make atomic changes**: Changes should be atomic, i.e. each pull request should only fix one issue or implement one feature. As such, each pull request should ideally only contain a single commit with an concise commit message.
```

Comment:
Too much emphasis on one commit per PR. Proposal: `Changes should be atomic. As such, pull requests should contain only few commits, and each commit should only fix one issue or implement one feature, with an concise commit message.`